### PR TITLE
Cloud functions update

### PR DESCRIPTION
**Pull request by John42506176Linux**
[
Screenshot 2020-08-05 at 10 36 58 AM
](url)
 Cloud Functions and dates

### CODE REVIEW

File: src/components/firebase-access.jsx [hunk 1]
```diff
@@ -101,6 +102,10 @@ export const initDates = async (symbols, Rounds) => {
 
     // No more than 7 rounds(Periods are measured in months)
     let min_window_size = 3;
+    if(IPOyearMax < 2000)
+    {
+      IPOyearMax = 2000 // Just to make sure there aren't any issue
+    }
```
Comment (asam01) [inline src/components/firebase-access.jsx]: please use proper style

if () {
  ....
}

File: src/components/firebase-access.jsx [hunk 2]
```diff
@@ -121,93 +126,130 @@ export const initDates = async (symbols, Rounds) => {
...
+  var formData = new FormData();
+  formData.append('symbol',JSON.stringify(symbols));
+  formData.append('RoomId',roomId);
+  formData.append('end-date',JSON.stringify(endDates));
+  var token =  localStorage.getItem('Token');
+  try{
+      let response = await fetch("/get_prices", {
+        method: 'POST',
+        body: formData,
+        //Comment this line back in when you want to deploy , and get rid of localhost
```
Comment (asam01) [inline src/components/firebase-access.jsx]: ?? delete?

File: src/components/firebase-access.jsx [hunk 3]
```diff
@@ -121,93 +126,130 @@ export const initDates = async (symbols, Rounds) => {
...
+  }
+  catch(err) {
+      console.log("Error is " +  err)
+  }
+  formData.append('periodLen',periodLen)
+  try{
+      let response  = await fetch('/get_stock_image', {
+          method: 'POST',
+          body: formData,
+          //Comment this line back in when you want to deploy , and get rid of localhost
```
Comment (asam01) [inline src/components/firebase-access.jsx]: ?

File: src/components/firebase-access.jsx [hunk 4]
```diff
@@ -121,93 +126,130 @@ export const initDates = async (symbols, Rounds) => {
...
+            Authorization:("Bearer " + token)
+          }
+      },100000)
+      console.log(response); // Same as above
+  }
+  catch(error){
+      console.log("Error is " +  error)
+  }
 }
+export const initSymbols = async(industry,Sector,MarketCap,NumOfSymbols) =>{
```
Comment (asam01) [inline src/components/firebase-access.jsx]: style-wise, really shouldn't be using capitalized letters for variable names. also, add a space after the =>

File: src/components/firebase-access.jsx [hunk 5]
```diff
@@ -121,93 +126,130 @@ export const initDates = async (symbols, Rounds) => {
...
+  if(atLeastTwo(industry,Sector,MarketCap)){
+      //In the case where a person wants to query using multiple features it may return a larger request,
+      //so we send it to the backend
+      let formData = new FormData();
+      if(industry)
+          formData.append('Industry',industry);
+      if(Sector)
+          formData.append('Sector',Sector);
+      if(MarketCap)
+          formData.append('MarketCap',MarketCap);
```
Comment (asam01) [inline src/components/firebase-access.jsx]: google style guides say to always use braces, even for one liners

File: src/components/firebase-access.jsx [hunk 6]
```diff
@@ -121,93 +126,130 @@ export const initDates = async (symbols, Rounds) => {
...
+      formData.append('NumOfSymbols',NumOfSymbols);
+      var token =  localStorage.getItem('Token');
+      try{
```
Comment (asam01) [inline src/components/firebase-access.jsx]: add spaces before braces. not just here but everywhere

File: src/components/firebase-access.jsx [hunk 7]
```diff
@@ -121,93 +126,130 @@ export const initDates = async (symbols, Rounds) => {
...
+          let response = await fetch('/get_symbols ', {
+              method: 'POST',
+              body: formData,
+              //Comment this line back in when you want to deploy , and get rid of localhost
```
Comment (asam01) [inline src/components/firebase-access.jsx]: delete

File: src/components/firebase-access.jsx [hunk 8]
```diff
@@ -121,93 +126,130 @@ export const initDates = async (symbols, Rounds) => {
...
-            .where("RandomPos",">=", cutoff)
-            .orderBy("RandomPos").limit(NumOfSymbols).get();
-        Stocks.forEach(function(Stock){
-            symbols.push(Stock.data().Symbol)
-        })
-    }
-    return symbols
+  }
+  // All of the rest of the values use a search by Xpos to make the query as small as possible(O(numOfSymbols))
+  else if(industry !== null){
```
Comment (asam01) [inline src/components/firebase-access.jsx]: space before brace

File: src/components/firebase-access.jsx [hunk 9]
```diff
@@ -121,93 +126,130 @@ export const initDates = async (symbols, Rounds) => {
...
+
+      let industryInfo =  await db.collection("Ticker-Info").doc("Industry").get();
+      let numOfIndustries= industryInfo.data().Industry[industry];
+      let cutoff = Math.floor((Math.random()  * (numOfIndustries - NumOfSymbols))+NumOfSymbols);
+      let industries = await db.collection("Ticker-Info").doc("Stock").collection("Stocks")
+          .where("Industry","==",industry)
+          .where("IndustryPos","<=", cutoff)
+          .orderBy("IndustryPos").limit(NumOfSymbols).get();
+      industries.forEach(function(doc){
```
Comment (asam01) [inline src/components/firebase-access.jsx]: add whitespace (blank line before 209)

File: src/components/firebase-access.jsx [hunk 10]
```diff
@@ -226,26 +268,45 @@ function randomDate(start, end) {
...
+//         initDates(symbolsL,rounds).then((datesD)=> {
+//             roomRef.update({
+//                 symbols: symbolsL,
+//                 dates: datesD["dates"],
+//             });
+//             initializeQuiz(symbolsL,roomID,datesD["period"],datesD["dates"]);
+//         });
+//     });
+//     return roomID;
+// }
```
Comment (asam01) [inline src/components/firebase-access.jsx]: ? delete

File: src/components/firebase-access.jsx [hunk 11]
```diff
@@ -226,26 +268,45 @@ function randomDate(start, end) {
...
+
+export const setUpRoom = (symbolsL,Rounds,password,startingMoney = 10000) => {
+  const roomRef = db.collection('Rooms').doc();
+  roomRef.set({
+      day_index: 0,
+      phase: 'no-host',
+      password: password,
+      starting_money: startingMoney,
```
Comment (asam01) [inline src/components/firebase-access.jsx:284]: be consistent with indentation. you've done 4 spaces on this one, but two spaces in the line before

File: src/components/firebase-access.jsx [hunk 12]
```diff
@@ -226,26 +268,45 @@ function randomDate(start, end) {
...
+  });
+  const roomID = roomRef.id;
+  initDates(symbolsL,Rounds).then((datesD)=> {
```
Comment (asam01) [inline src/components/firebase-access.jsx]: space before =>

Comment (asam01) [reply src/components/firebase-access.jsx]: also please use better variable names than 'symbolsL', it's unclear what 'L' stands for

File: src/components/firebase-access.jsx [hunk 13]
```diff
@@ -417,7 +478,7 @@ function compDoc(a, b){
 }
 
 export const getIndustries = async (db) => {
-  let Industries = await db.collection("Ticker-Info").doc("Industry").get();
+  let Industries = await db.collection("Ticker-Info").doc("Indust").get();
```
Comment (asam01) [inline src/components/firebase-access.jsx]: ? was this intentional?

File: src/components/pages/Host.jsx [hunk 14]
```diff
@@ -139,7 +136,7 @@ class Host extends Component {
                 roomRef.update({
                     phase: this.state.phase
                 })
-                await addUser(roomId,"dummy user");
+                // await addUser(roomId,"dummy user");
```
Comment (asam01) [inline src/components/pages/Host.jsx]: delete

File: src/components/quiz/create/Config.jsx [hunk 15]
```diff
@@ -32,10 +31,15 @@ class Config extends React.Component {
...
-        this.props.updatePageType('created',roomID);
+      const Industry = null;
+      const Sector = null;
+      const MarketCaps = null;
+      const {numSymbols,numRounds} = this.state;
+
+      initSymbols(Industry,Sector,MarketCaps,numSymbols).then((symbolsL) => {
+            let roomID = setUpRoom(symbolsL,numRounds,'');
+            this.props.updatePageType('created',roomID);
+      });
```
Comment (asam01) [inline src/components/quiz/create/Config.jsx:42]: I'm not sure that we agreed on adding this back in the day of the demo... @jack-y-xu ?

File: src/firebase.js [hunk 16]
```diff
@@ -37,12 +37,20 @@ const db = createDB(USE_LOCALHOST);
 
 const provider = new firebase.auth.GoogleAuthProvider();
 const signInWithGoogle = () => {
-  auth.signInWithPopup(provider);
+  auth.signInWithPopup(provider).then(function(result) {
+    // This gives you a Google Access Token. You can use it to access the Google API.
+    var token = result.credential.idToken;
+    console.log('Token is ' + token);
```
Comment (asam01) [inline src/firebase.js]: delete

File: stock_backend/IntrinsicValue.py [hunk 17]
```diff
@@ -7,11 +7,11 @@
 import pandas as pd
 import matplotlib.pyplot as plt
 # Uncomment this when we start using earnings
-# from yahoo_earnings_calendar import YahooEarningsCalendar 
+# from yahoo_earnings_calendar import YahooEarningsCalendar
```
Comment (asam01) [inline stock_backend/IntrinsicValue.py]: delete

File: stock_backend/IntrinsicValue.py [hunk 18]
```diff
@@ -30,12 +30,12 @@ def getLatestDividend(symbol):
 #TODO: Make get earnings calendar so we can add alerts for the game
 # def getEarningsDate()
 '''
-Paramaters: 
-Data-A pandas dataframe of the Closing data of a stock 
+Paramaters:
```
Comment (asam01) [inline stock_backend/IntrinsicValue.py]: fix spelling on all of these - 'parameters'

**Review by asam01 — CHANGES_REQUESTED** [review]
remove all files from the py_cache as well. those don't need to be on here

File: firestore.rules [hunk 19]
```diff
@@ -0,0 +1,17 @@
+
+// Allow read/write access on all documents to any user signed in to the application
+//service cloud.firestore {
+//  match /databases/{database}/documents {
+//    match /{document=**} {
+//      allow read, write: if request.auth != null;
```
Comment (asam01) [inline firestore.rules]: wouldn't we want this in? we are the only ones who should be able to read from and write to db

File: src/components/firebase-access.jsx (hunk 2 above)
Comment (John42506176Linux) [reply src/components/firebase-access.jsx]: ?? why

Comment (asam01) [reply src/components/firebase-access.jsx]: there's nothing to comment in?

File: firestore.rules [hunk 20]
```diff
@@ -1,13 +1,4 @@
...
-//  match /databases/{database}/documents {
-//    match /{document=**} {
-//      allow read, write: if request.auth != null;
-//    }
-//  }
-//}
-
 service cloud.firestore {
   match /databases/{database}/documents {
     match /{document=**} {
```
Comment (asam01) [inline firestore.rules]: again, don't we want to allow read/write only if authorization is not null?

File: src/components/firebase-access.jsx [hunk 21]
```diff
@@ -131,11 +130,11 @@ export const initializeQuiz = async (symbols, roomId, periodLen, endDates) => {
...
-  try{
+  try {
       let response = await fetch("/get_prices", {
         method: 'POST',
         body: formData,
-        //Comment this line back in when you want to deploy , and get rid of localhost
+        //Comment this line back out when you want to run on localhost and add localhost, and comment out headers
         headers: {
           Authorization: ("Bearer " + token)
         },
```
Comment (asam01) [inline src/components/firebase-access.jsx]: like I mentioned before, can you just add a bool at the top of the file `useLocalhost = false` and then modify these HTTP requests to align with whatever the bool value is?

File: src/components/firebase-access.jsx (hunk 8 above)
Comment (asam01) [reply src/components/firebase-access.jsx]: still applies

File: src/components/firebase-access.jsx [hunk 22]
```diff
@@ -196,42 +198,46 @@ export const initSymbols = async(industry,Sector,MarketCap,NumOfSymbols) =>{
...
 
   }
+
   // All of the rest of the values use a search by Xpos to make the query as small as possible(O(numOfSymbols))
   else if(industry !== null){
 
       let industryInfo =  await db.collection("Ticker-Info").doc("Industry").get();
       let numOfIndustries= industryInfo.data().Industry[industry];
-      let cutoff = Math.floor((Math.random()  * (numOfIndustries - NumOfSymbols))+NumOfSymbols);
+      let cutoff = Math.floor((Math.random()  * (numOfIndustries - numOfSymbols))+numOfSymbols);
```
Comment (asam01) [inline src/components/firebase-access.jsx]: be consistent with spacing
) + numSymbols
add spaces surrounding the `+`

File: src/components/firebase-access.jsx [hunk 23]
```diff
@@ -196,42 +198,46 @@ export const initSymbols = async(industry,Sector,MarketCap,NumOfSymbols) =>{
...
-          .orderBy("MarketCapPos").limit(NumOfSymbols).get();
-      Stocks.forEach(function(doc){
+          .orderBy("MarketCapPos").limit(numOfSymbols).get();
+
+      stocks.forEach(function(doc){
           symbols.push(doc.data().Symbol);
       })
   }
-  else if(Sector !== null){
+  else if(sector !== null){
```
Comment (asam01) [inline src/components/firebase-access.jsx]: space before {

File: src/components/firebase-access.jsx [hunk 24]
```diff
@@ -196,42 +198,46 @@ export const initSymbols = async(industry,Sector,MarketCap,NumOfSymbols) =>{
...
 
-      let SectorInfo =  await db.collection("Ticker-Info").doc("Sector").get();
-      let numOfSectors= SectorInfo.data().Sector[Sector];
-      let cutoff = Math.floor((Math.random()  * (numOfSectors - NumOfSymbols))+NumOfSymbols);
-      let Sectors = await db.collection("Ticker-Info").doc("Stock").collection("Stocks")
-          .where("Sector","==",Sector)
+      let sectorInfo =  await db.collection("Ticker-Info").doc("Sector").get();
+      let numOfSectors= sectorInfo.data().Sector[sector];
```
Comment (asam01) [inline src/components/firebase-access.jsx]: style?
numOfSectors = sectorInfo
use space surrounding `=`

File: src/components/firebase-access.jsx [hunk 25]
```diff
@@ -240,10 +246,11 @@ export const initSymbols = async(industry,Sector,MarketCap,NumOfSymbols) =>{
 
       let StockInfo =  await db.collection("Ticker-Info").doc("Stock").get();
       let numOfStocks = StockInfo.data().NumOfStocks - 1;
-      let cutoff = Math.floor((Math.random()  * (numOfStocks - NumOfSymbols))+NumOfSymbols);
+      let cutoff = Math.floor((Math.random()  * (numOfStocks - numOfSymbols))+numOfSymbols);
```
Comment (asam01) [inline src/components/firebase-access.jsx]: same here for ` + `

**Review by asam01 — COMMENTED** [review]
still a few things. left comments on specific areas.

File: src/components/firebase-access.jsx (hunk 21 above)
Comment (John42506176Linux) [reply src/components/firebase-access.jsx]: That's a more intensive commit, right now. That can be handled later.

File: firestore.rules (hunk 20 above)
Comment (asam01) [reply firestore.rules]: can you reply before marking as resolved?

File: firestore.rules [hunk 26]
```diff
@@ -1,8 +1,10 @@
 
+Allow read/write access on all documents to any user signed in to the application
```
Comment (asam01) [inline firestore.rules]: is this supposed to be commented? did this compile?

**Review by asam01 — COMMENTED** [review]
one last thing

**Comment by John42506176Linux** [comment]
@googlebot I signed it!

**Comment by John42506176Linux** [comment]
@googlebot I signed it!

**Review by asam01 — APPROVED** [review]

**Review by asam01 — CHANGES_REQUESTED** [review]
looked back at all the files. there are still a few unresolved comments that you had resolved without making the changes.

also, be sure to delete pycache files and push the deletion. right now they still appear

**Review by asam01 — APPROVED** [review]

**Review by yukyhaha — APPROVED** [review]

**Comment by John42506176Linux** [comment]
@googlebot I signed it!

**Pull request merged**